core/client/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -40,20 +44,37 @@
 public final class AlluxioBlockStore {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
-  private static AlluxioBlockStore sClient = null;
+  // Only one block store will be kept for each master.
+  private static final Map<InetSocketAddress, AlluxioBlockStore> CACHED_BLOCK_STORES =
```

Comment:
I suppose essentially that's the same as with the current change: the AlluxioBlockStore members in different objects are referencing the same piece of AlluxioBlockStore instance if the master is the same. The AlluxioBlockStore instances are cached in the static map to guarantee uniqueness. With this change, it makes it possible to talk to different AlluxioBlockStores within one JVM (although there's such logic in the current code)